squidpy/im/_container.py

```diff
@@ -349,23 +351,25 @@ def transform_metadata(data: xr.Dataset) -> xr.Dataset:
 
             return data
 
-        img = Path(img)
-        logg.debug(f"Loading data from `{img}`")
-
-        if not img.exists():
-            raise OSError(f"Path `{img}` does not exist.")
+        is_url = validators.url(str(img_path_or_url))
+        img = Path(img_path_or_url)
+        if not is_url:
+            if not img.exists():
+                raise OSError(f"Path `{img}` does not exist.")
 
         suffix = img.suffix.lower()
 
         if suffix in (".jpg", ".jpeg", ".png", ".tif", ".tiff"):
             return _lazy_load_image(img, dims=dims, chunks=chunks)
 
-        if img.is_dir():
+        if img.is_dir() or suffix == ".zarr":
             if len(self._data):
                 raise ValueError("Loading data from `Zarr` store is disallowed when the container is not empty.")
+            data_str = img_path_or_url if is_url else str(img)
+            self._data = transform_metadata(xr.open_zarr(data_str, chunks=chunks))
+            return
```

Comment:
`mypy` for some reason doesn't like this, would just use `if/elif` instead.
Also, could you please modify the `raise ValueError(f"Unable to handle path `{img_path}`.")` on line `378`. since we're no longer dealing with suffixes; this may break some tests.